src/Scanner/BucketsManager.ts

```diff
@@ -0,0 +1,50 @@
+import { WatchableBucket } from './WatchableBucket';
+import { Bucket } from '../Buckets';
+import { BucketWatchCallback } from './BucketWatchCallback';
+import { WatchableBucketFactory } from './WatchableBucketFactory';
+import { ILogger, DefaultLogger } from '../Logger';
+
+export class BucketsManager {
+  private buckets: WatchableBucket[] = [];
+  private watchableBucketFactory: WatchableBucketFactory;
+  private logger: ILogger;
+
+  constructor(
+    watchableBucketFactory: WatchableBucketFactory,
+    logger: ILogger = new DefaultLogger()
+  ) {
+    this.watchableBucketFactory = watchableBucketFactory;
+    this.logger = logger;
+  }
+
+  public async stop() {
+    await Promise.all(this.buckets.map(b => b.stop()));
+    return;
```

Comment:
No need for `return` here. It's implicit.

```suggestion
```

Reply:
It's needed, as I don't want to return Promise<void[]>